src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Schemas/CatalogDiscountType.cs

```diff
@@ -12,13 +16,19 @@ public class CatalogDiscountType : DiscountType
     {
         public CatalogDiscountType(IMediator mediator, IDataLoaderContextAccessor dataLoader)
         {
-            Field<PromotionType, Promotion>()
-                .Name("promotion")
-                .ResolveAsync(ctx =>
+            var promotion = new EventStreamFieldType
+            {
+                Name = "promotion",
+                Type = GraphTypeExtenstionHelper.GetActualType<PromotionType>(),
+                Arguments = new QueryArguments(),
+                Resolver = new AsyncFieldResolver<Discount, Promotion>(async ctx =>
                 {
                     var loader = dataLoader.Context.GetOrAddBatchLoader<string, Promotion>("promotionsLoader", (ids) => LoadPromotionsAsync(mediator, ids));
-                    return loader.LoadAsync(ctx.Source.PromotionId);
-                });
+                    var result = loader.LoadAsync(ctx.Source.PromotionId);
+                    return await result.GetResultAsync();
```

Comment:
As I remember this syntax won't work for batchLoaders you must ` return loader.LoadAsync(ctx.Source.PromotionId);` without `await`

Reply:
how does it work?

![Screenshot_2](https://user-images.githubusercontent.com/20122385/111467516-3062ee00-872d-11eb-9e91-ace4fd3a045d.png)